### PR TITLE
feat: allow distribution to be passed through SsrSite

### PR DESCRIPTION
**Pull request by khuezy**
fixes: #3022
- Refactored SsrSite to removed a lot of redundant code
- Enhance `serverCachePolicy` to be `ICachePolicy` or an object of `headers, query, or cookies` string[]
- Allow NextjsSite cdk.distribution to pass a cdk Distribution

I haven't tested this locally via `pnpm link`, this is just to get early feedback from @fwang

### CODE REVIEW

File: packages/sst/src/constructs/SsrSite.ts [hunk 1]
```diff
@@ -1053,14 +1029,30 @@ function handler(event) {
...
-        allowedHeaders && allowedHeaders.length > 0
-          ? CacheHeaderBehavior.allowList(...allowedHeaders)
+        headers && headers.length > 0
+          ? CacheHeaderBehavior.allowList(...new Set([...headers, ...headersProps]))
           : CacheHeaderBehavior.none(),
-      cookieBehavior: CacheCookieBehavior.all(),
+      cookieBehavior:
+        cookies && cookies.length > 0
+          ? CacheCookieBehavior.allowList(...new Set([...cookies, ...cookieProps]))
+          : CacheCookieBehavior.all(),
```
Comment (khuezy) [inline packages/sst/src/constructs/SsrSite.ts:1055]: This should be default to `.none()`

**Comment by jayair** [comment]
Relevant Discord thread https://discord.com/channels/983865673656705025/1121404657978310707/1124158746315202620

**Comment by liam-reilly** [comment]
Is this likely to be merged anytime soon?

**Comment by thdxr** [comment]
We are transition this repo to https://github.com/sst/v2 and are closing this issue for now. If it is still relevant please reopen it there - sorry for the trouble

**Pull request closed without merging**